pkg/envoy/xds_server.go

```diff
@@ -199,10 +207,20 @@ func toAny(pb proto.Message) *anypb.Any {
 }
 
 // newXDSServer creates a new xDS GRPC server.
-func newXDSServer(envoySocketDir string, ipCache IPCacheEventSource, localEndpointStore *LocalEndpointStore) (*xdsServer, error) {
+func newXDSServer(envoySocketDir string, ipCache IPCacheEventSource, localEndpointStore *LocalEndpointStore,
+
+	proxyGID int, httpRequestTimeout int, httpIdleTimeout int, httpMaxGRPCTimeout int, httpRetryCount int, httpRetryTimeout int, httpNormalizePath bool,
```

Comment:
Thanks for pointing that out @tklauser . I agree that this is quite hard to read.

At first i thought about passing the newly introduced Hive Config struct `envoyProxyConfig` into the "constructors" - but i decided against it because this way we might pass too many unnecessary information. They are quite coarse grained because they are used by multiple components within the cell. In general i tent to keep the Hive components (config, lifecycle, ...) as close as possible to the cell itself. but that's just my opinion.

alternatives:
- additional "fine-grain" config struct per "constructor" (e.g. `xdsServerConfig`) (only for the config values - not the depencencies (IPCache, ...))
- options pattern (for the config values - not the dependencies (IPCache, ...))
- custom Go types for the config values (e.g. `proxyGID`). This would to use the types for documentation purposes in the tests, even though it would be technically still possible to pass the plain values.
- use comments in the tests (`/*proxyGID*/ 1337, /*httpRequestTimeout*/ 0, ...`)

If it's OK i would go with a dedicated "fain-grain" config struct? this also comes with the nice benefit of having the dependencies "separated" from the config (instead of the long parameter list)

Reply:
Thanks @mhofstetter for elaborating in detail!

> At first i thought about passing the newly introduced Hive Config struct `envoyProxyConfig` into the "constructors" - but i decided against it because this way we might pass too many unnecessary information. They are quite coarse grained because they are used by multiple components within the cell. In general i tent to keep the Hive components (config, lifecycle, ...) as close as possible to the cell itself. but that's just my opinion.

Makes sense, I didn't check the `envoyProxyConfig` struct close enough to notice that it is more coarse grained.

> If it's OK i would go with a dedicated "fain-grain" config struct? this also comes with the nice benefit of having the dependencies "separated" from the config (instead of the long parameter list)

:+1: That would be my preferred option too.

Reply:
I introduced explicit config structs for the following functions with long parameter lists.

- `newXdsServer`
- `startEmbeddedEnvoy`
- `writeBootstrapConfigFile`

@tklauser PTAL

Reply:
hmm it would even be possible to keep the config as field of the xdsServer struct itself. let me refactor this

Reply:
@tklauser done